kotlinx-coroutines-core/common/src/channels/Produce.kt

```diff
@@ -25,6 +25,32 @@ public interface ProducerScope<in E> : CoroutineScope, SendChannel<E> {
     val channel: SendChannel<E>
 }
 
+/**
+ * Suspends the current coroutine until the channel is either [closed][SendChannel.close] or [cancelled][ReceiveChannel.cancel]
+ * and invokes the given [block] before resuming the coroutine.
```

Comment:
Additional documentation would be helpful here: 

"Note that when producer channel is cancelled this function resumes with cancellation exception, so putting the code after call this function would not lead to its execution in case of cancellation. That is why this code takes a lambda parameter."